File: Documentação/Documentação/Template Latex - Relatorio Tecnico - IFSP - SBV/fontes/Pygame/rotate.py
def rotate(self):
    if self.num == 7:
        return
    newBlocks = []
    for block in self.blocks:
        newX = -block.y
        newY = block.x
        newBlocks.append(Vector2(newX, newY))
        
    if self.isValidMove(Vector2(0,0), newBlocks):
        self.blocks = newBlocks
        return newBlocks
    elif self.isValidMove(Vector2(1,0), newBlocks):
        self.position.x += 1
        self.blocks = newBlocks
        return newBlocks
    elif self.isValidMove(Vector2(-1,0), newBlocks):
        self.position.x -= 1
        self.blocks = newBlocks
        return newBlocks
    elif self.isValidMove(Vector2(0,1), newBlocks):
        self.position.y += 1 
        self.blocks = newBlocks
        return newBlocks
    return self.blocks

File: Documentação/Documentação/Template Latex - Relatorio Tecnico - IFSP - SBV/fontes/Raylib/rotate.cpp
void Piece::rotate(){
    if(num == 4){
        return;
    }
    Vector2 newBlocks[4];
    for(int i = 0; i < 4; i++){
        newBlocks[i].x = -blocks[i].y;
        newBlocks[i].y = blocks[i].x;
    }
    if(isValidMove({0,0}, newBlocks)){
        for(int i = 0; i < 4; i++){
            blocks[i] = newBlocks[i];
        }
        return ;
    }
    if(isValidMove({1,0}, newBlocks)){
        for(int i = 0; i < 4; i++){
            blocks[i] = newBlocks[i];
        }
        position.x += 1;
        return ;
    } 
    if(isValidMove({-1,0}, newBlocks)){
        for(int i = 0; i < 4; i++){
            blocks[i] = newBlocks[i];
        }
        position.x -= 1;
        return;
    } 
    if(isValidMove({0,1}, newBlocks)){
        for(int i = 0; i < 4; i++){
            blocks[i] = newBlocks[i];
        }
        position.y += 1;
        return;
    }
}